cli/tests/integration_tests.rs

```diff
@@ -412,6 +412,80 @@ fn js_unit_tests() {
   assert!(status.success());
 }
 
+#[test]
+fn ts_dependency_recompilation() {
+  // First we have to generate a bundle of some module that has exports.
+  let mod1 = util::root_path().join("cli/tests/subdir/mod1.ts");
+  assert!(mod1.is_file());
+  let t = TempDir::new().expect("tempdir fail");
+  let ats = t.path().join("a.ts");
+
+  std::fs::write(
+    &ats,
+    "
+    import { foo } from \"./b.ts\";
+
+    function print(str: string): void {
+        console.log(str);
+    }
+    
+    print(foo);",
+  )
+  .expect("error writing file");
+
+  let bts = t.path().join("b.ts");
+  std::fs::write(
+    &bts,
+    "
+    export const foo = \"foo\";",
+  )
+  .expect("error writing file");
+
+  let output = util::deno_cmd()
+    .current_dir(util::root_path())
+    .env("NO_COLOR", "1")
+    .arg("run")
+    .arg(&ats)
+    .output()
+    .expect("failed to spawn script");
+
+  let stdout_output = std::str::from_utf8(&output.stdout).unwrap().trim();
+  let stderr_output = std::str::from_utf8(&output.stderr).unwrap().trim();
+
+  assert!(stdout_output.ends_with("foo"));
+  assert!(stderr_output.starts_with("Compile"));
+
+  // Overwrite contents of b.ts and run again
+  std::fs::write(
+    &bts,
+    "
+    export const foo = 5;",
+  )
+  .expect("error writing file");
+
+  let output = util::deno_cmd()
+    .current_dir(util::root_path())
+    .env("NO_COLOR", "1")
+    .arg("run")
+    .arg(&ats)
+    .output()
+    .expect("failed to spawn script");
+
+  let stdout_output = std::str::from_utf8(&output.stdout).unwrap().trim();
+  let stderr_output = std::str::from_utf8(&output.stderr).unwrap().trim();
+
+  let err_lines = stderr_output
+    .split('\n')
+    .map(|s| s.to_string())
+    .collect::<Vec<_>>();
+  assert!(err_lines.get(0).unwrap().starts_with("Compile"));
+  assert_eq!(
+    err_lines.get(1).unwrap(),
+    "error: TS2345 [ERROR]: Argument of type '5' is not assignable to parameter of type 'string'."
+  );
+  assert_eq!(stdout_output, "");
+}
```

Comment:
I've checked that this test fails on master. Nice work!

Consider making the asserts here a bit more robust to change. I think you could get away with a single assert
```rust
assert!(stderr_output.contains("TS2345 [ERROR]"));
```